controlplane/pkg/sidecars/dns_nsm_monitor_handler.go

```diff
@@ -0,0 +1,53 @@
+package sidecars
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/networkservicemesh/k8s/cmd/nsm-coredns/env"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/networkservicemesh/k8s/cmd/nsm-coredns/api/update"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+)
+
+//nsmDNSMonitorHandler implements NSMMonitorHandler interface for handling dnsConfigs
+type nsmDNSMonitorHandler struct {
+	EmptyNSMMonitorHandler
+	dnsConfigUpdateClient update.DNSConfigServiceClient
+}
+
+//NewNsmDNSMonitorHandler creates new DNS monitor handler
+func NewNsmDNSMonitorHandler() NSMMonitorHandler {
+	clientSock := env.UpdateAPIClientSock.StringValue()
+	if clientSock == "" {
+		logrus.Fatal("update api server socket not passed")
```

Comment:
It is not informative to exit with such message, please add logging.

Reply:
thanks, fixed